src/device/rpi3/uarts.go

```diff
@@ -0,0 +1,213 @@
+// +build rpi3
+
+package rpi3
```

Comment:
This file should be in the machine package: src/machine/machine_rpi3.go.
Also, the API is very different from other boards. Take a look here for example:

https://github.com/tinygo-org/tinygo/blob/cc313798f381532e4874a7166fa3c216234c005d/src/machine/machine_fe310.go#L33-L54

In general we try to follow existing interfaces, such as `io.Writer`, `io.Reader`, and if needed we copy the signature of similar objects like the [WriteByte](https://golang.org/pkg/bufio/#Writer.WriteByte) method.